cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/web/BaseOAuthWrapperController.java

```diff
@@ -106,24 +107,28 @@ protected OAuthWebApplicationService createService(final RegisteredService regis
      * Create an authentication from a user profile.
      *
      * @param profile the given user profile
+     * @param service the registered service
      * @return the built authentication
      */
-    protected Authentication createAuthentication(final UserProfile profile) {
```

Comment:
Sure thing that it might not be feasible today, but I'd second Misagh's sentiment that this code duplication might "smell". I think we need to think hard how to accommodate this use case and refactor the internal/external API and flow to achieve that. We have the luxury of the major release here so we could make any numbers of necessary breaking changes and we need to take advantage of it.